src/jit/flowgraph.cpp

```diff
@@ -14527,8 +14534,7 @@ bool Compiler::fgOptimizeUncondBranchToSimpleCond(BasicBlock* block, BasicBlock*
     assert(block->bbJumpKind == BBJ_ALWAYS);
     assert(block->bbJumpDest == target);
 
-    // TODO-Review: OK if they are in the same region?
-    if (compHndBBtabCount > 0)
+    if (!BasicBlock::sameEHRegion(block, target))
```

Comment:
that is the second change, it gives us a bigger code size (because now we are cloning more).

Reply:
Much of the time the cloned and original blocks can be optimized away, and so code size improves.